### PR TITLE
SC super vendor

### PR DESCRIPTION
**Pull request by Jinyu-W**
# Description

- Add SuperStorageUnit to support super vendor
- Add super_vendor topology
- Add a notebook for SCI data processing
- Fix world build, env sampler, price sample etc. accordingly

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [ ] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [ ] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [ ] Windows
  - [ ] Mac OS
  - [ ] Linux
- Python version:
  - [ ] 3.6
  - [ ] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: maro/simulator/scenarios/supply_chain/topologies/core.yml [hunk 1]
```diff
@@ -36,6 +36,9 @@ units:
           # class: class name, joined with the module path to find the class type
           class: "StorageUnit"
           datamodel: "StorageDataModel"
+        SuperStorageUnit:
+          class: "SuperStorageUnit"
+          datamodel:
```
Comment (lihuoran) [inline maro/simulator/scenarios/supply_chain/topologies/core.yml:41]: Leave it empty or use `None` / an empty string?

File: maro/simulator/scenarios/supply_chain/sku_dynamics_sampler.py [hunk 2]
```diff
@@ -200,18 +200,18 @@ def _init_info_dict(self) -> Dict[str, DynamicsInfoItem]:
         }
 
     def _sample_attr(self, tick: int, product_id: int, attr_name: str) -> object:
-        if any([
-            tick not in self._cache,
-            product_id not in self._cache[tick],
-            attr_name not in self._cache[tick][product_id],
-        ]):
+        if (
```
Comment (lihuoran) [inline maro/simulator/scenarios/supply_chain/sku_dynamics_sampler.py:203]: Wondering if we should add a customized `any` function with short-circuiting.

File: maro/simulator/scenarios/supply_chain/units/storage.py [hunk 3]
```diff
@@ -327,3 +327,53 @@ def get_unit_info(self) -> StorageUnitInfo:
             **super(StorageUnit, self).get_unit_info().__dict__,
             product_list=[i for i in self._product_level.keys()],
         )
+
+
+class SuperStorageUnit(UnitBase):
```
Comment (lihuoran) [inline maro/simulator/scenarios/supply_chain/units/storage.py:332]: If `SuperStorageUnit` provides identical interfaces with `StorageUnit` and there is no inheritance relationship between them, I suggest adding a higher abstraction to define the interfaces. For example, `StorageUnit` => `SimpleStorageUnit` / `SuperStorageUnit`.

File: maro/simulator/scenarios/supply_chain/units/storage.py [hunk 4]
```diff
@@ -327,3 +327,53 @@ def get_unit_info(self) -> StorageUnitInfo:
...
+        facility: FacilityBase, parent: Union[FacilityBase, UnitBase], world: World, config: dict
+    ) -> None:
+        super(SuperStorageUnit, self).__init__(id, data_model_name, data_model_index, facility, parent, world, config)
+
+    def initialize(self) -> None:
+        super(SuperStorageUnit, self).initialize()
+
+    @property
+    def capacity(self) -> int:
+        raise NotImplementedError
```
Comment (lihuoran) [inline maro/simulator/scenarios/supply_chain/units/storage.py:344]: Return `None` or raise error? Same for the following methods.

File: maro/simulator/scenarios/supply_chain/units/unitbase.py [hunk 5]
```diff
@@ -123,7 +123,7 @@ def get_unit_info(self) -> BaseUnitInfo:
         return BaseUnitInfo(
             id=self.id,
             node_index=self.data_model_index,
-            node_name=type(self.data_model).__node_name__,
+            node_name=type(self.data_model).__node_name__ if self.data_model else None,
```
Comment (lihuoran) [inline maro/simulator/scenarios/supply_chain/units/unitbase.py:126]: Change the type of `BaseUnitInfo.node_name` from `str` to `Optional[str]`.

File: maro/simulator/scenarios/supply_chain/topologies/core.yml (hunk 1 above)
Comment (Jinyu-W) [reply maro/simulator/scenarios/supply_chain/topologies/core.yml:41]: change to "null"

**Pull request merged**